neuraxle/pipeline.py

```diff
@@ -55,13 +60,73 @@ def inverse_transform(self, processed_outputs):
         return self.inverse_transform_processed_outputs(processed_outputs)
 
 
+class PipelineRepository(ABC):
+    def __init__(self, cache_folder, pipeline_cache_list_file_name='pipeline_cache_list.txt'):
+        self.pipeline_cache_list_file_name = pipeline_cache_list_file_name
+        self.cache_folder = cache_folder
+
+    def save(self, pipeline: 'Pipeline', data_container: DataContainer):
```

Comment:
I realize that you often name methods the same way in objects that doesn't inherit from the same interface (e.g.: here and in previous PRs). Although I name reusing the same terms where applicable, I would avoid naming methods from different OOP contracts the same way.